### ISSUE TITLE
Android version not work on Samsung S6

### ISSUE THREAD

**Issue by sarovin** [marcshilling/react-native-image-picker#229]
Hi,
i have this component:

```
import React, {
  PropTypes,
  Component,
} from 'react';
import {
  Platform,
  StyleSheet,
  View,
  Text,
  Image,
  TouchableOpacity,
} from 'react-native';
import Colors from '../Styles/Colors';
import Fonts from '../Styles/Fonts';
import ImagePicker from 'react-native-image-picker';

const styles = StyleSheet.create({
  container: {
    margin: 16,
    alignItems: 'center',
  },
  circle: {
    width: 95,
    height: 95,
    borderRadius: 95 / 2,
    backgroundColor: Colors.colorBgTwo,
    borderWidth: 13,
    borderColor: 'white',
    alignItems: 'center',
    justifyContent: 'center',
  },
  avatar: {
    width: 82,
    height: 82,
    borderRadius: 82 / 2,
  },
});

const propTypes = {
  options: PropTypes.object,
};

const defaultProps = {
  options: {
    title: 'Select Avatar', // specify null or empty string to remove the title
    cancelButtonTitle: 'Cancel',
    takePhotoButtonTitle: 'Take Photo...', // specify null or empty string to remove this button
    chooseFromLibraryButtonTitle: 'Choose from Library...', // specify null or empty string to remove this button
    cameraType: 'front', // 'front' or 'back'
    mediaType: 'photo', // 'photo' or 'video'
    quality: 0.5, // 0 to 1, photos only
    angle: 0, // android only, photos only
    allowsEditing: false, // Built in functionality to resize/reposition the image after selection
    noData: false, // photos only - disables the base64 `data` field from being generated (greatly improves performance on large photos)
    storageOptions: { // if this key is provided, the image will get saved in the documents directory on ios, and the pictures directory on android (rather than a temporary directory)
      skipBackup: true, // ios only - image will NOT be backed up to icloud
      path: 'images', // ios only - will save image at /Documents/images rather than the root
    },
  },
};

class ImageUpload extends Component {

  constructor(props) {
    super(props);

    this.state = {
      avatarSource: null,
    };
    this.selectPhotoTapped = this.selectPhotoTapped.bind(this);
  }

  selectPhotoTapped() {
    ImagePicker.showImagePicker(this.props.options, (response) => {
      console.log('Response = ', response);

      if (response.didCancel) {
        console.log('User cancelled photo picker');
      } else if (response.error) {
        console.log('ImagePicker Error: ', response.error);
      } else if (response.customButton) {
        console.log('User tapped custom button: ', response.customButton);
      } else {
        var source;

        // You can display the image using either:
        // source = { uri: `data:image/jpeg;base64, ${response.data}`, isStatic: true };
        if (Platform.OS === 'android') {
          source = { uri: response.uri, isStatic: true };
        } else {
          source = { uri: response.uri.replace('file://', ''), isStatic: true };
        }

        this.setState({
          avatarSource: source,
        });
      }
    });
  }

  render() {
    return (
      <View style={styles.container}>
        <TouchableOpacity onPress={() => { this.selectPhotoTapped(); }} >
          <View style={styles.circle}>
            {this.state.avatarSource === null ? <Text style={Fonts.h7ButtonPrimary}>Photo</Text> :
              <Image style={styles.avatar} source={this.state.avatarSource} />
            }
          </View>
        </TouchableOpacity>
      </View>
    );
  }
}

ImageUpload.propTypes = propTypes;
ImageUpload.defaultProps = defaultProps;

export default ImageUpload;

```
On IOS it works, on Android i have this:

![img_2016-06-13 13 33 32](https://cloud.githubusercontent.com/assets/5980398/16006097/7a2bd640-316b-11e6-83c0-907671f1e555.jpg)

Can you help me?

**Comment by yfuks** [comment]
I run the Example and i don't get this issue. Can you try the Example and tell me if it work for you ?
Are you sure that you use the last version of this package ?

**Comment by sarovin** [comment]
@yfuks yes, i have react-native-image-picker@0.19.2.
Something went wrong with the installation...so i have deleted android folder and i have reinstalled the component:
```
react-native android
rnpm link react-native-image-picker 
react-native run-android
```

but i have this error:
```
Unknown source file : UNEXPECTED TOP-LEVEL EXCEPTION:
Unknown source file : com.android.dex.DexException: Multiple dex files define Landroid/support/v7/appcompat/R$anim;
Unknown source file : 	at com.android.dx.merge.DexMerger.readSortableTypes(DexMerger.java:596)
Unknown source file : 	at com.android.dx.merge.DexMerger.getSortedTypes(DexMerger.java:554)
Unknown source file : 	at com.android.dx.merge.DexMerger.mergeClassDefs(DexMerger.java:535)
Unknown source file : 	at com.android.dx.merge.DexMerger.mergeDexes(DexMerger.java:171)
Unknown source file : 	at com.android.dx.merge.DexMerger.merge(DexMerger.java:189)
Unknown source file : 	at com.android.dx.command.dexer.Main.mergeLibraryDexBuffers(Main.java:502)
Unknown source file : 	at com.android.dx.command.dexer.Main.runMonoDex(Main.java:334)
Unknown source file : 	at com.android.dx.command.dexer.Main.run(Main.java:277)
Unknown source file : 	at com.android.dx.command.dexer.Main.main(Main.java:245)
Unknown source file : 	at com.android.dx.command.Main.main(Main.java:106)

:app:dexDebug FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:dexDebug'.
> com.android.ide.common.process.ProcessException: org.gradle.process.internal.ExecException: Process 'command '/Library/Java/JavaVirtualMachines/jdk1.8.0_74.jdk/Contents/Home/bin/java'' finished with non-zero exit value 2

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED
```

**Comment by yfuks** [comment]
Well since something went wrong i recommend you to link the module manualy (steps are in the README)
I don't know any of the files in your stack error.
Make sure you updated **RN**

**Comment by sarovin** [comment]
@yfuks 

After 

> rnpm link react-native-image-picker

 i have found this in settings.gradle
```
rootProject.name = 'Example'

include ':app'
include ':react-native-image-picker'
project(':react-native-image-picker').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-image-picker/android')
```

but in README there is this
```
rootProject.name = 'Example'

include ':app'
include ':react-native-image-picker'
project(':react-native-image-picker').projectDir = new File(settingsDir, '../node_modules/react-native-image-picker/android')
```

and in AndroidManifest.xml there isn't this:```

<uses-permission android:name="android.permission.CAMERA" />
    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE"/>
    <uses-feature android:name="android.hardware.camera" android:required="false"/>
    <uses-feature android:name="android.hardware.camera.autofocus" android:required="false"/>
```

I want fix it but I do not know how rnpm works...

**Comment by yfuks** [comment]
I don't know either. If the Example work fine for you try to find all the diff with your project.

**Comment by deming1990** [comment]
i have same problem, you need to delete node_modules/react-native-image-picker/android/build/, it's works

**Comment by sarovin** [comment]
@deming1990 Can you send a PR?

**Comment by deming1990** [comment]
after you run ```sh npm install react-native-image-picker@latest --save ```

node_modules/react-native-image-picker/android/build/

delete original /android/build/

@sarovin

**Comment by sarovin** [comment]
Is there another way? Can we fix it in this repo?

**Comment by marcshilling** [comment]
I'm able to reproduce. Investigating...

**Comment by marcshilling** [comment]
Are you sure you followed the README instructions? What you're seeing is exactly what would happen if you failed to link the native project. I just `react-native init` a new project, `npm install react-native-image-picker@latest`, followed the steps to link the native library, and it built and launched the picker fine.

**Comment by marcshilling** [comment]
Should be fixed in `0.19.5`. Let me know.

**Comment by theLallanTop** [comment]
Hi,

> android {
>     compileSdkVersion 23
>     buildToolsVersion "23.0.2"
>      defaultConfig {
>         applicationId "com.XXX"
>         minSdkVersion 16
>         targetSdkVersion 22
>         versionCode 1
>         versionName "1.0"
>         ndk {
>             abiFilters "armeabi-v7a", "x86"
>         }
>     }
> 

> "react-native-image-picker": "^0.22.1"

,
it's work fine other device but not with Samsung galaxy tab, s3
